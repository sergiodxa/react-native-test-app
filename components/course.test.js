import React from 'react';
import Course from './course';

import renderer from 'react-test-renderer';

describe('Course item', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(
      <Course
        badge="https://static.platzi.com/media/achievements/programacion-basica2x.png"
        name="Programaci칩n b치sica"
      />
    ).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders the expected UI', () => {
    const rendered = renderer.create(
      <Course
        badge="https://static.platzi.com/media/achievements/programacion-basica2x.png"
        name="Programaci칩n b치sica"
      />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
