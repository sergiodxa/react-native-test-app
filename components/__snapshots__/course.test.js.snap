// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course item renders the expected UI 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://static.platzi.com/media/achievements/programacion-basica2x.png",
      }
    }
    style={
      Array [
        Object {
          "height": 36,
          "marginRight": 8,
          "width": 36,
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Programación básica
  </Text>
</View>
`;
